src/QtLocationPlugin/GoogleMapProvider.cpp

```diff
@@ -105,37 +108,28 @@ void GoogleMapProvider::_tryCorrectGoogleVersions(
         QSslConfiguration conf = qheader.sslConfiguration();
         conf.setPeerVerifyMode(QSslSocket::VerifyNone);
         qheader.setSslConfiguration(conf);
-        QString url = "http://maps.google.com/maps/api/js?v=3.2&sensor=false";
+        const QString url = QStringLiteral("http://maps.google.com/maps/api/js?v=3.2&sensor=false");
         qheader.setUrl(QUrl(url));
         QByteArray ua;
         ua.append(getQGCMapEngine()->userAgent());
         qheader.setRawHeader("User-Agent", ua);
         _googleReply = networkManager->get(qheader);
-        connect(_googleReply, &QNetworkReply::finished, this,
-                &GoogleMapProvider::_googleVersionCompleted);
-        connect(_googleReply, &QNetworkReply::destroyed, this,
-                &GoogleMapProvider::_replyDestroyed);
-        connect(
-            _googleReply,
-            static_cast<void (QNetworkReply::*)(QNetworkReply::NetworkError)>(
-                &QNetworkReply::error),
-            this, &GoogleMapProvider::_networkReplyError);
+        connect(_googleReply, &QNetworkReply::finished, this, &GoogleMapProvider::_googleVersionCompleted);
+        connect(_googleReply, &QNetworkReply::destroyed, this, &GoogleMapProvider::_replyDestroyed);
+        connect(_googleReply, QOverload<QNetworkReply::NetworkError>::of(&QNetworkReply::error), this, &GoogleMapProvider::_networkReplyError);
         networkManager->setProxy(proxy);
     }
 }
 
-QString
-GoogleStreetMapProvider::_getURL(int x, int y, int zoom,
-                                 QNetworkAccessManager* networkManager) {
+QString GoogleStreetMapProvider::_getURL(const int x, const int y, const int zoom, QNetworkAccessManager* networkManager) {
     // http://mt1.google.com/vt/lyrs=m
-    QString server  = "mt";
-    QString request = "vt";
-    QString sec1    = ""; // after &x=...
-    QString sec2    = ""; // after &zoom=...
+    QString server  = QStringLiteral("mt");
+    QString request = QStringLiteral("vt");
+    QString sec1; // after &x=...
```

Comment:
Why removing the initialization ? Could be QStringLiteral("") ?
What is the motivation of passing everything in QStringLiteral ?

Reply:
- There is no need for initialization of statically allocated `QString`  objects because the default constructor will be called anyways giving you [an empty and null QString](https://doc.qt.io/qt-5/qstring.html#QString). To have the initialization would cost us some extra processing in the QString internals which is not needed since in `GoogleMapProvider::_getSecGoogleWords` they get wiped again. 
- When you pass a `char[]` to methods that take `QString` as argument a temporary `QString` is created and that requires some memory allocations and conversions in the implementation of `QString` that are quite expensive. You can see this if you do some performance profiling using QtCreator, especially on the maps/terrain data part where we build all those urls for each tile.